app/uk/gov/hmrc/cipphonenumberverification/services/VerifyHelper.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cipphonenumberverification.services
+
+import play.api.Logging
+import play.api.libs.json.Json
+import play.api.mvc.Results.{Accepted, BadRequest, InternalServerError, Ok}
+import uk.gov.hmrc.cipphonenumberverification.connectors.GovUkConnector
+import uk.gov.hmrc.cipphonenumberverification.models.{ErrorResponse, IndeterminateResponse, Passcode, PhoneNumber, VerificationStatus}
+import uk.gov.hmrc.cipphonenumberverification.repositories.PasscodeCacheRepository
+import uk.gov.hmrc.http.HttpReads.{is2xx, is4xx}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.mongo.cache.DataKey
+
+import java.security.SecureRandom
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+abstract class VerifyHelper(passcodeCacheRepository: PasscodeCacheRepository, govUkConnector: GovUkConnector) extends Logging {
```

Comment:
rename file as i believe there should be a separate file for db stuff and a separate file for GovNotify helper methods.  the otp should also be in its own helper or util class, as it could be reused else where and has nothing to do with db calls or GovNotify.  Theres too much going on in this class basically.

Reply:
- Why not have a DAO class for the database methods, persist passcode, get passcode etc.
- Have a GovNotify connector class for the GovNotify stuff.  
- and a Service class to orchestrate it all.